src/modules/Library/features/AppletsCatalog/AppletsCatalog.test.tsx

```diff
@@ -0,0 +1,164 @@
+    },
+    {
+      id: '4e76faca-45d1-4bf6-9c2f-26572aa79c8a',
+      displayName: 'Applet Share Test 6',
+      description: {
+        en: 'Description',
+      },
+      image: '',
+      keywords: ['library'],
+      activities: [],
+      version: '1.1.0',
+    },
+  ],
+};
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useNavigate: () => mockUseNavigate,
+}));
+
+jest.mock('redux/store/hooks', () => ({
+  ...jest.requireActual('redux/store/hooks'),
+  useAppDispatch: jest.fn(),
+}));
+
+jest.mock('modules/Library/hooks', () => ({
+  ...jest.requireActual('modules/Library/hooks'),
+  useAppletsFromCart: jest.fn(),
+  useReturnToLibraryPath: jest.fn(),
+}));
+
+describe('AppletsCatalog', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.spyOn(reduxHooks, 'useAppDispatch').mockReturnValue(mockDispatch);
+    jest.spyOn(library.thunk, 'getPublishedApplets').mockReturnValue(() => {});
+  });
+
+  test('renders loading spinner when applets are loading', () => {
+    const spyGetPublishedApplets = jest.spyOn(library.thunk, 'getPublishedApplets');
+    jest.spyOn(library, 'usePublishedApplets').mockReturnValue({ count: 0, result: [] });
+    jest.spyOn(library, 'usePublishedAppletsStatus').mockReturnValue('loading');
+    jest.spyOn(library, 'useCartAppletsStatus').mockReturnValue('loading');
+
+    renderWithProviders(<AppletsCatalog />);
+
+    expect(screen.getByTestId('spinner')).toBeInTheDocument();
+    expect(spyGetPublishedApplets).toHaveBeenCalledWith({ limit: 6, page: 1, search: '' });
+  });
+
+  test('renders applets and pagination correctly when applets are loaded', async () => {
+    const spyGetPublishedApplets = jest.spyOn(library.thunk, 'getPublishedApplets');
+    jest.spyOn(library, 'usePublishedApplets').mockReturnValue(mockPublishedApplets);
+    jest.spyOn(library, 'usePublishedAppletsStatus').mockReturnValue('idle');
+    jest.spyOn(library, 'useCartAppletsStatus').mockReturnValue('idle');
+
+    renderWithProviders(<AppletsCatalog />);
+
+    expect(screen.queryByTestId('spinner')).not.toBeInTheDocument();
+    expect(spyGetPublishedApplets).toHaveBeenCalledWith({ limit: 6, page: 1, search: '' });
+
+    expect(screen.queryAllByTestId(/library-applets-\d+$/)).toHaveLength(6);
+    expect(screen.getByTestId('library-applets-pagination')).toBeInTheDocument();
+
+    expect(screen.getByText('1–6 of 16'));
+    const nextPageButton = screen.getByRole('button', { name: 'Go to next page' });
+    await userEvent.click(nextPageButton);
+
+    expect(spyGetPublishedApplets).toHaveBeenCalledWith({ limit: 6, page: 2, search: '' });
+
+    const searchContainer = screen.getByTestId('library-search');
+    expect(searchContainer).toBeInTheDocument();
+    const searchInput = searchContainer.querySelector('input');
+
+    await userEvent.type(searchInput, 'share test 1');
+    await new Promise((resolve) => setTimeout(resolve, SEARCH_DEBOUNCE_VALUE));
```

Comment:
Let's reduce time consuming by using jest functions to work with time:
https://jestjs.io/docs/timer-mocks
please use `useFakeTimers`  (and `advanceTimersByTime` in this case) to control timers in tests.